### PR TITLE
partyChain

### PR DESCRIPTION
**Pull request by paulmarrington**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -49,4 +49,73 @@ To add an identity:
...
+    "sharingAgencyIds" : [
+    ]
+
+### List Relationship for a Party
+
+    GET http://host/api/1/Relationship/List/delegate/{PartyId}/page/{pageNo/size/{itemsPerPage}
+
+e.g.
+
+    GET http://host/api/1/Relationship/List/570c866b64457f7c32907800/delegate/{PartyId}/page/0/size/50
```
Comment (sundriver) [inline docs/api.md]: doesn't match the sample

File: backend/typescript/controllers/Relationship.ts [hunk 2]
```diff
@@ -23,23 +24,21 @@ export function RelationshipAPI() {
 
     /* list relationships for a specific delegate party */
     router.get(
-    "/List/:delegate_or_subject/:value/:type/page/:page/size/:pagesize",
+    "/List/:delegate_or_subject/:_id/page/:page/size/:pagesize",
```
Comment (sundriver) [inline backend/typescript/controllers/Relationship.ts]: url casing always lower case

Comment (sundriver) [inline backend/typescript/controllers/Relationship.ts]: How can I specify /relationship/relId1/relId2/...?

File: backend/typescript/controllers/Relationship.ts [hunk 3]
```diff
@@ -23,23 +24,21 @@ export function RelationshipAPI() {
...
+      var query: { [key: string] : any } = { deleted: false }
+      query[delegate_or_subject + "PartyId"] =
+      new mongoose.Types.ObjectId(req.params._id)
+      model.find(query)
+      .skip((req.params.page - 1) * req.params.page_size)
+      .limit(req.params.page_size)
+      .lean()
+      .find(function(err: any, relDocs: IRelationship[]) {
+        if (!err) {
+          res.json(relDocs)
```
Comment (sundriver) [inline backend/typescript/controllers/Relationship.ts]: Response must be wrapped in IReponse

File: backend/typescript/controllers/Relationship.ts [hunk 4]
```diff
@@ -75,6 +74,66 @@ export function RelationshipAPI() {
         }
       })
     });
+    
+    /* Provided navigation details for a relationship */
+    router.get(
+    "/Table/:delegate_or_subject/:_id/page/:page/size/:pagesize",
```
Comment (sundriver) [inline backend/typescript/controllers/Relationship.ts]: Casing of URL

File: backend/typescript/controllers/Relationship.ts [hunk 5]
```diff
@@ -75,6 +74,66 @@ export function RelationshipAPI() {
...
+        var ident = pd.identities[0]
+        next({
+          id:       identityId,
+          name:     ident.name,
+          subName:  (ident.type === "abn") ? ident.value : ""
+        })
+      })
+  }
+  
+  router.get("/Path/*", (req, res) => {
```
Comment (sundriver) [inline backend/typescript/controllers/Relationship.ts]: this implies subject id can be optional, which is not the case
Correct: /path/:identity/:relationshipIds?*

File: backend/typescript/controllers/Relationship.ts [hunk 6]
```diff
@@ -75,6 +74,66 @@ export function RelationshipAPI() {
...
+      relIds.forEach((relId, idx) => {
+        model.findById(relId, (err: any, relDoc: IRelationship) => {
+          var nickname = relDoc.subjectsNickName.split("//")
+          navRes.partyChain.push({
+            id:       relId,
+            name:     nickname[0],
+            subName:  nickname[1]
+          })
+          if (idx === (relIds.length - 1)) {
+            res.json(navRes)
```
Comment (sundriver) [inline backend/typescript/controllers/Relationship.ts]: Wrapping in IResponse

File: docs/api.md [hunk 7]
```diff
@@ -49,4 +49,73 @@ To add an identity:
...
+
+e.g.
+
+    GET http://host/api/1/Relationship/570b2f98f7d1d96813bdd456
+    
+Returns:
+    "_id" : "570c866b64457f7c32907806", 
+    "updatedAt" : ISODate("2016-04-12T05:23:55.087+0000"), 
+    "createdAt" : ISODate("2016-04-12T05:23:55.087+0000"), 
+    "type" : "Business",
```
Comment (sundriver) [inline docs/api.md]: I can't consume type:"Business" so no reason to include it here

File: backend/typescript/controllers/Relationship.ts (hunk 2 above)
Comment (paulmarrington) [inline backend/typescript/controllers/Relationship.ts]: I was following Lachlan's where he has upper case. Will change

File: backend/typescript/controllers/Party.ts [hunk 8]
```diff
@@ -35,7 +35,7 @@ export function PartyAPI() {
    */
   router.post("/", (req, res) => {
     model.create(req.body,
-    function(err: any, partyDoc: IParty) {
+    (err: any, partyDoc: IParty) => {
       if (err) {
         res.status(500).send(err.toString());
```
Comment (sundriver) [inline backend/typescript/controllers/Party.ts]: IResponsify the error message please

File: backend/typescript/controllers/Relationship.ts (hunk 2 above)
Comment (sundriver) [inline backend/typescript/controllers/Relationship.ts]: Good point, I just checked ATO REST Guidelines, the want CamelCase on URLs, I change on my side.

Comment (sundriver) [inline backend/typescript/controllers/Relationship.ts]: One wiki it says: The first letter of a camel case compound word may or may not be capitalized
Now that they start the url with api that potentially mean thisIsCorrect while ThisIsNotClear

**Pull request merged**